### PR TITLE
[WNMGDS-2861] Adds Email, Link, RSS & What's New Icon components

### PR DESCRIPTION
**Pull request by tamara-corbalt**
## Summary

Adds four new icon components:
- `EmailIcon`
- `LinkIcon`
- `RssIcon`
- `WhatsNewIcon`

See Jira ticket for details: [WNMGDS-2861](https://jira.cms.gov/browse/WNMGDS-2861). Please note that a [Download icon](https://github.com/CMSgov/design-system/blob/main/packages/design-system/src/components/Icons/DownloadIcon.tsx) has already been added to the codebase.

## How to test

1. Run `yarn storybook`.
2. Navigate to the Icons section and verify the presence and appearance of the new icons.

## Checklist

- [X] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [X] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [X] Selected appropriate `Impacts`, multiple can be selected.
- [X] Selected appropriate release milestone

Screenshot 2024-08-14 at 3 19 39 PM
Screenshot 2024-08-14 at 3 19 57 PM
Screenshot 2024-08-14 at 3 20 07 PM
Screenshot 2024-08-14 at 3 20 23 PM
Screenshot 2024-08-14 at 3 20 49 PM

### CODE REVIEW

**Review by malloryiden — APPROVED** [review]
🎉

**Review by jack-ryan-nava-pbc — APPROVED** [review]
LGTM! Question about the viewBox control/params: we don't have any controls set up in Storybook, should we?

**Comment by tamara-corbalt** [comment]
@jack-ryan-nava-pbc I like the idea of supporting viewBox controls because it gives Storybook users the flexibility to customize the icons. However, I'm concerned about the added complexity to our Storybook configuration and the potential for misuse. Could changing the proportions of the icons potentially distort the layout of the icon table?

File: packages/design-system/src/components/Icons/EmailIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import type * as React from 'react';
+import { t } from '../i18n';
+import { SvgIcon, IconCommonProps } from './SvgIcon';
+
+const defaultProps = {
+  className: '',
+  viewBox: '0 0 32 32',
```
Comment (zarahzachz) [inline packages/design-system/src/components/Icons/EmailIcon.tsx:7]: Nothing wrong here, just a callout that it's interesting the `viewBox` is the same for all new icons. I wonder if there's an opportunity to standardize our icons in the future?

File: packages/design-system/src/components/locale/en.json [hunk 2]
```diff
@@ -49,22 +49,26 @@
...
     "remove": "Remove",
+    "rss": "RSS",
     "star": "Star",
     "starEmpty": "Empty Star",
     "starFilled": "Filled Star",
     "starHalf": "Half Star",
     "usaLogo": "USA.gov - Government Made Easy",
-    "warning": "Warning"
+    "warning": "Warning",
+    "whatsNew": "What's New"
```
Comment (kim-cmsds) [inline packages/design-system/src/components/locale/en.json:71]: nit: replace this quote with one that sparks joy for typography enthusiasts
```suggestion
    "whatsNew": "What’s New"
```

**Review by kim-cmsds — APPROVED** [review]

File: packages/design-system/src/components/locale/en.json (hunk 2 above)
Comment (tamara-corbalt) [reply packages/design-system/src/components/locale/en.json:71]: @kim-cmsds I love the idea of sparking joy for typography enthusiasts. Will make the change.

Comment (tamara-corbalt) [reply packages/design-system/src/components/locale/en.json:71]: @kim-cmsds, I made the change to use a curly apostrophe (’), but I'm noticing a tiny box appearing around it in vs code. I'm not sure what this signifies or if it indicates a potential issue with the character encoding? 
Screenshot 2024-08-14 at 12 51 00 PM

Comment (kim-cmsds) [reply packages/design-system/src/components/locale/en.json:71]: That same curly apostrophe is used here:
https://github.com/CMSgov/design-system/blob/0fb69bcaa6f1505e09db7c00349c939b1c37222f/packages/design-system/src/components/locale/en.json#L108

Comment (tamara-corbalt) [reply packages/design-system/src/components/locale/en.json]: @kim-cmsds: updated to a curly apostrophe in this json file, as well as `icons.stories.tsx`

**Review by kim-cmsds — APPROVED** [review]

**Pull request merged**